### ISSUE TITLE
slicing passmanagers

### ISSUE THREAD

**Issue by ajavadia** [Qiskit/qiskit-terra#3195]
following #3004, implement methods to slice PassManagers to only get a subset of passes:

```python
pm = PassManager(...)

pm[1]
pm[1:3]

pm1[1:3] + pm2[4:6]  # be able to concatenate several passmanagers
pm_level1[:8] + pm_3rdparty[0] + pm_level1[9:]
```

**Comment by faisaldebouni** [comment]
I'm trying to implement this. However I need some guidance:
`pm[..]` should return a `PassManager` with subset of passes (rather than the subset of passes it self) **correct?**
because this way it would allow to carry conditions, options .. etc
also it will allow something like 
`pm[0:3].draw()`
it will also allow
`pm[..] + pm[..]`
and
`pm + pm[..]`
to act in similar way.

the only downside would be that this won't work:
`pm1[3] = pm2[1]
`

**Comment by 1ucian0** [comment]
> pm[..] should return a PassManager with subset of passes (rather than the subset of passes it self) correct?

Yes. It should. Currently, a PassManager does not fully emulate a container type. And I think it should though. A PassManager is a collection of pass sets (`FlowController` instances) that are saved in `working_list` list. That's the list that is important here.

"Wrap" the `working_list`. Let me know if that makes sense. We can discuss it with code.

**Comment by faisaldebouni** [comment]
That makes perfect sense. Thanks. Will report once I finish.

**Comment by 1ucian0** [comment]
pending PR in #3209

**Comment by 1ucian0** [comment]
closed via #3209
